web-marketplace/src/ledger.tsx

```diff
@@ -88,11 +88,12 @@ export const LedgerProvider: React.FC<React.PropsWithChildren<unknown>> = ({
   const [api, setApi] = useState<RegenApi | undefined>(undefined);
   const [loading, setLoading] = useState<boolean>(false);
   const [error, setError] = useState<unknown>(undefined);
-  const { wallet } = useWallet();
+  const { wallet, loaded } = useWallet();
 
   useEffect(() => {
-    getApi(setApi, setLoading, setError, wallet?.offlineSigner);
-  }, [setApi, setLoading, setError, wallet?.offlineSigner]);
+    if (loaded && !api)
```

Comment:
now only connecting to ledger RegenApi once, avoiding a few json rpc queries with method "status"